### PR TITLE
[4.3] Converts editor buttons to service providers

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes

- Move all editor buttons except page break (for testing purposes) to service providers

### Testing Instructions

- Apply the PR
- Delete the file `administrator/cache/autoload_psr4.php`
- Check that all buttons work correctly for all the editors (tinyMCE, Codemirror, None)

### Actual result BEFORE applying this Pull Request

Works

### Expected result AFTER applying this Pull Request

Works

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

@laoneo

### CODE REVIEW

File: plugins/editors-xtd/article/article.xml [hunk 1]
```diff
@@ -9,8 +9,10 @@
 	<authorUrl>www.joomla.org</authorUrl>
 	<version>3.0.0</version>
 	<description>PLG_ARTICLE_XML_DESCRIPTION</description>
+	<namespace path="src">Joomla\Plugin\EditorsXtd\Article</namespace>
 	<files>
-		<filename plugin="article">article.php</filename>
+		<folder>services</folder>
+		<folder plugin="article">src</folder>
```
Comment (laoneo) [inline plugins/editors-xtd/article/article.xml]: Normally we added the plugin attribute to the service folder.

Comment (dgrammatiko) [reply plugins/editors-xtd/article/article.xml]: done

**Comment by laoneo** [comment]
I'm just wondering if the namespace should be a bit more clear. I know they are inline with the rest, but Xtd doesn't say much. Why not just going with Joomla\Plugin\EditorButton\Article.

**Comment by dgrammatiko** [comment]
Let’s keep the Xtd suffix, as I plan to roll a new set with the upcoming modal (dialog element) and @Fedik wants to have couple more events for the editors so the EditorsButtons would be used there

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on 9f0987df77d8308c94a9d2b087aab1d4cbf52d75<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39649">issues.joomla.org/tracker/joomla-cms/39649</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39649">issues.joomla.org/tracker/joomla-cms/39649</a>.</sub>

**Comment by obuisard** [comment]
Thank you Dimitris @dgrammatiko :-)

**Pull request merged**